### ISSUE TITLE
Bring in 64 bit compiled ET

### ISSUE THREAD

**Issue by cipriangal** [JeffersonLab/japan#16]

**Comment by cipriangal** [comment]
Bob says:
```
Here's what I remember doing for the feedback / ET

1. Needed a BLOCKING station (but ultimately dont want that)
THaEtClient.C:  et_station_config_setblock(sconfig, ET_STATION_BLOCKING);

2. I obtained a version of ET that worked, see the Makefile in
  ~/PREX/feedback/pan/codaclass26

specifically
# these are sources used by CODA 2.6.2.
LIBETDIR = /adaqfs/home/apar/et-12.0/src/libsrc/.Linux-x86_64/

3  ./codaclass32 -- trying to compile with -m32, does not work
                     because needs ROOT and syslibs with -m32
(Unless you want all old 32-bit libs and recompile ROOT with
32-bit ... ugh!)

4. Made various attempts with CODA 2.5 but it did not work.
We are stuck with CODA 2.6.2 unless we upgrade the frontends
from vxWorks boards to Intel PCs.  Then we could use CODA 3
and the ET would work more transparently.
```
And also:
```
I recall that the main trick was to pretend that the ET system was "remote".
This way, you use sockets to get the data, and become immune to the problem
of hooking up to an apparently 32-bit system.

  et_open_config_setmode(openconfig, ET_HOST_AS_REMOTE);  
  // et_open_config_setmode(openconfig, ET_HOST_AS_LOCAL);

The trick was suggested by Carl Timmer, who is the author of the ET system.

Also note that we link against ET version 12 and use those header files.
That was the version underlying CODA 2.6.2.
```

**Comment by cipriangal** [comment]
Bob further says:
```
 Use ET version 12.  The main trick to get ET to work and avoid "et ERROR: etl_open: ET system is 32 bit and this program is 64 bit!" is to open the ET station as "remote".  et_open_config_setmode(openconfig, ET_HOST_AS_REMOTE); instead of ET_HOST_AS_LOCAL.  While that gets it to work, Carl said "Setting the config to 'as remote' should slow things down. Removing it should make it work much faster. All it means is you're forcing local communication to use sockets instead of shared memory."  This might be why we had to run the station in blocking mode to get all the events.  We should talk to Carl some more, and possibly get more support for this.
```

**Comment by paulmking** [comment]
The pking.qwfeedback branch does now have the ability to connect to the 32-bit complied ET server using the 64-bit compiled library and sockets as described above.

**Comment by cipriangal** [comment]
got pan to connect to the ET system as per Bob's instructions above. However there is a significant performance hit with going through a socket. Next step is to try to get the DAQ group to compile ET12 under 64 bit

**Comment by cipriangal** [comment]
Bob says:
Idea from DAQ group: cannot compile the primary ET as 64-bit b/c would need to compile all of CODA that way, too much work. Instead make a 2nd ET system (64-bit) connect to the primary ET (32-bit) using a blocking socket connection.  Then clients like JAPAN connect as  non-blocking station to the 64-bit ET.  This should have the desired performance.  Carl Timmer to provide this 64-bit ET and example code.  Bob will test the performance with our DAQ.

**Comment by rwmichaels** [comment]
A functioning 2-ET system was developed and the minor changes to THaEtClient was submitted (pulled into) the development branch.  So, this is done, though we may still want to tweak it.  Here is some documentation:  ttps://userweb.jlab.org/~rom/TwoET.html

**Comment by cameronc137** [comment]
[See halog 3645013](https://logbooks.jlab.org/entry/3645013)

I managed to use the 64 bit ET system in online analysis mode, but the root file output produced was empty and improperly named.

**Comment by paulmking** [comment]
Using the "online" flag (plus the ET options) in branch bugfix-online-epicscheck can now connect to the dual ET system and properly decode events.

**Comment by cipriangal** [comment]
Sounds great .. when you create the pull request can you put in detailed instructions on how to test it?

**Comment by paulmking** [comment]
Resolved on develop by pull #76 .

**Comment by cameronc137** [comment]
Carl Timmer has issued a bug fix for 64 bit ET in CODA 2.6.2 that allows for the ETbridge to properly treat it as remote. I've installed these on the atedf3 test stand and proved that the bugfix works. Now we don't need to wait 20 seconds for the et bridge to come on line at start and can just trust it to work from startcoda itself.

**Comment by cameronc137** [comment]
I fixed this in apar@adaq3 today https://logbooks.jlab.org/entry/3663543#comment-20592
